validator_client/src/block_service.rs

```diff
@@ -585,6 +761,49 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
         Ok::<_, BlockError>(())
     }
 
+    async fn get_validator_block_v3(
+        beacon_node: &BeaconNodeHttpClient,
+        slot: Slot,
+        randao_reveal_ref: &SignatureBytes,
+        graffiti: Option<Graffiti>,
+        proposer_index: Option<u64>,
+        builder_boost_factor: Option<u64>,
+        log: &Logger,
+    ) -> Result<UnsignedBlock<E>, BlockError> {
+        let (block_response, _) = beacon_node
+            .get_validator_blocks_v3::<E>(
+                slot,
+                randao_reveal_ref,
+                graffiti.as_ref(),
+                builder_boost_factor,
+            )
+            .await
+            .map_err(|e| {
+                BlockError::Irrecoverable(format!(
```

Comment:
I think this should also be recoverable, but it doesn't make much difference because the v3 logic doesn't do any retries.

Reply:
Done in https://github.com/sigp/lighthouse/pull/4813/commits/f579a043067eec26072baba2bf3a8505c33f8159